From 2950685774c5987145aae55adde18bab182ab7a7 Mon Sep 17 00:00:00 2001
From: Aseem Kumar <aseemk@google.com>
Date: Wed, 11 Jun 2025 18:41:20 -0700
Subject: [PATCH] Don't allow SdkSandbox to bypass systemUid check.

Test: atest

Flag: EXEMPT security fix

Bug: 397438392
Change-Id: I1003ac2a795c869aaeb292692b2681c7a5c8d12e
(cherry picked from commit d400709160c8374d83a15dc7623b11434c08c4c6)
---
 .../com/android/server/accounts/AccountManagerService.java     | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/core/java/com/android/server/accounts/AccountManagerService.java b/services/core/java/com/android/server/accounts/AccountManagerService.java
index faebe7359eb5..c2c1a64ac2d3 100644
--- a/services/core/java/com/android/server/accounts/AccountManagerService.java
+++ b/services/core/java/com/android/server/accounts/AccountManagerService.java
@@ -5895,6 +5895,9 @@ public class AccountManagerService
     }
 
     private boolean isSystemUid(int callingUid) {
+        if (Process.isSdkSandboxUid(callingUid)) {
+            return false;
+        }
         String[] packages = null;
         final long ident = Binder.clearCallingIdentity();
         try {
-- 
2.50.1.351.gcf940e82a1-goog

